### PR TITLE
Replace uses of `add_scaled_row!` by `addmul!`

### PR DESCRIPTION
**Pull request by lgoettgens**
to make the code easier understandable (the argument order of `add_scaled_row!` is kind of unintuitive).

When @HechtiDerLachs wrote the code, he already wanted to use `addmul!` but this was not available back then.

cc @HechtiDerLachs

### CODE REVIEW

File: experimental/DoubleAndHyperComplexes/src/Morphisms/simplified_complexes.jl [hunk 1]
```diff
@@ -496,7 +496,7 @@ function _simplify_matrix!(A::SMat; find_pivot=nothing)
     # clear the q-th column
     for (i, b) in a_col_del
       #A[i] = A[i] - uinv*b*a_row # original operation, replaced by inplace arithmetic below
-      Hecke.add_scaled_row!(a_row, A[i], -uinv*b)
+      addmul!(A[i], a_row, -b*uinv)
```
Comment (thofma) [inline experimental/DoubleAndHyperComplexes/src/Morphisms/simplified_complexes.jl]: Why was `-uinv*b` changed?

Comment (lgoettgens) [reply experimental/DoubleAndHyperComplexes/src/Morphisms/simplified_complexes.jl]: Typo

Comment (lgoettgens) [inline experimental/DoubleAndHyperComplexes/src/Morphisms/simplified_complexes.jl]:
```suggestion
      addmul!(A[i], a_row, -uinv*b)
```

Comment (thofma) [reply experimental/DoubleAndHyperComplexes/src/Morphisms/simplified_complexes.jl]: ok, I thought it was some left/right thing

**Review by fingolfin — APPROVED** [review]

**Pull request merged**